packages/shared/src/components/streak/popup/DayStreak.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ReactElement } from 'react';
+import ReadingStreakIcon from '../../icons/ReadingStreak';
+import classed from '../../../lib/classed';
+import TriangleArrowIcon from '../../icons/Arrow/Triangle';
+import { IconSize } from '../../Icon';
+
+export enum Streak {
+  Completed = 'completed',
+  Pending = 'pending',
+  Upcoming = 'upcoming',
+  Freeze = 'freeze',
+}
+
+interface DayStreakProps {
+  streak: Streak;
+  day: number;
+  shouldShowArrow?: boolean;
+}
+
+const dayInitial = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
+const Circle = classed(
+  'div',
+  'w-7 h-7 rounded-full border border-theme-divider-tertiary',
+);
+
+export function DayStreak({
+  streak,
+  day,
+  shouldShowArrow,
+}: DayStreakProps): ReactElement {
+  const renderIcon = () => {
+    if (streak === Streak.Completed || streak === Streak.Pending) {
+      return (
+        <ReadingStreakIcon
+          secondary={streak === Streak.Completed}
+          size={IconSize.Medium}
+        />
+      );
+    }
+
+    return (
+      <Circle
+        className={streak === Streak.Freeze && 'bg-theme-label-disabled'}
```

Comment:
Technically you could have completed a freeze right?
(Might be an edge-case we need to tackle in development)
CC: @DragosIorgulescu

Reply:
Yes, that should be possible and should be handled by the consuming component (ReadingStreakPopup). This kind of dumb component where will only act based on props, and not necessarily contain the logic to decide which day is completed or not.